cypress/integration/patch-route.js

```diff
@@ -108,36 +109,81 @@ describe("Patch route", function() {
     });
 
     describe("Tasks Table", () => {
-      it("updates the url when column headers are clicked", () => {
+      beforeEach(() => {
+        cy.server();
+        cy.route("POST", "/graphql/query").as("gqlQuery");
+        cy.visit(path);
+      });
+
+      it("Updates the url when column headers are clicked", () => {
         cy.visit(path);
 
-        cy.get("th.cy-task-table-col-name").click();
-        locationHasUpdatedParams("NAME", 1);
+        cy.get("th.cy-task-table-col-NAME").click();
+        locationHasUpdatedParams("NAME", "ASC");
 
-        cy.get("th.cy-task-table-col-name").click();
-        locationHasUpdatedParams("NAME", -1);
+        cy.get("th.cy-task-table-col-NAME").click();
+        locationHasUpdatedParams("NAME", "DESC");
 
-        cy.get("th.cy-task-table-col-name").click();
+        cy.get("th.cy-task-table-col-NAME").click();
         locationHasUpdatedParams("NAME");
 
-        cy.get("th.cy-task-table-col-variant").click();
-        locationHasUpdatedParams("VARIANT", 1);
+        cy.get("th.cy-task-table-col-VARIANT").click();
+        locationHasUpdatedParams("VARIANT", "ASC");
 
-        cy.get("th.cy-task-table-col-variant").click();
-        locationHasUpdatedParams("VARIANT", -1);
+        cy.get("th.cy-task-table-col-VARIANT").click();
+        locationHasUpdatedParams("VARIANT", "DESC");
 
-        cy.get("th.cy-task-table-col-variant").click();
+        cy.get("th.cy-task-table-col-VARIANT").click();
         locationHasUpdatedParams("VARIANT");
       });
 
       it("clicking task name goes to task page for that task", () => {
         cy.visit(path);
-        cy.get("td.cy-task-table-col-name:first").within(() => {
+        cy.get("td.cy-task-table-col-NAME:first").within(() => {
           cy.get("a")
             .should("have.attr", "href")
             .and("include", "/task");
         });
       });
+
+      it("Should have sort buttons disabled when fetching data", () => {
+        cy.visit(path);
+        cy.contains(TABLE_SORT_SELECTOR, "Name").click();
+        cy.once("fail", err => {
+          expect(err.message).to.include(
+            "'pointer-events: none' prevents user mouse interaction."
+          );
+        });
+      });
+
+      it("Fetches sorted tasks when table sort headers are clicked", () => {
+        ["NAME", "STATUS"].forEach(sortBy =>
```

Comment:
only testing these two because it felt redundant (and expensive) to test all four statuses. these are the two statuses i subjectively deem to be "the most likely"

Reply:
I think testing the other sort headers is a useful test and not including them could mislead us. The code to handle moving query params from the event listener to the URL and query variables is specific and could break in such a way where NAME and STATUS pass but the other sort headers fail.

Reply:
i agree!